src/vs/platform/terminal/common/terminal.ts

```diff
@@ -391,7 +393,7 @@ export interface IShellLaunchConfig {
 	 * The codicon ID to use for this terminal. If not specified it will use the default fallback
 	 * icon.
 	 */
-	icon?: string;
+	iconPath?: ThemeIcon | Codicon | string | URI | { light: URI, dark: URI };
```

Comment:
Do we still need string here? If it's just the ID, we could replace its usage by `{ id: '...' }` to simplify this type?

Reply:
was trying to avoid having to fix terminal profiles in this PR, which have icon: string